common/constants/default.go

```diff
@@ -155,4 +154,7 @@ const (
 
 	// DefaultManifestsDir edge node default static pod path
 	DefaultManifestsDir = "/etc/kubeedge/manifests"
+
+	// EdgeHub Quic
+	DefaultQuicPort = 10001
```

Comment:
Looks good.
A small question, if we use the `DefaultQuicPort` above, so the port for quic will not be customizable. Now the `ip-port` is used to set the ip and port for cloudcore, how about add a parameter for `protocol`, then set the `ip-port` to websocket or quic?
cc @Shelley-BaoYue @WillardHu

Reply:
Yes, the configuration of edgecore can be changed after the join node to support quic. We can add some flags in the join command to support edgehub to choose websocket or quic.

Reply:
> Looks good. A small question, if we use the `DefaultQuicPort` above, so the port for quic will not be customizable. Now the `ip-port` is used to set the ip and port for cloudcore, how about add a parameter for `protocol`, then set the `ip-port` to websocket or quic? cc @Shelley-BaoYue @WillardHu

The default is to use websocket as the communication protocol. If you need to switch to the quic protocol or modify the port of quic, you need to modify the configmap and edge edgecore. yaml of the cloud cloud after the keadm join.

Reply:
> Yes, the configuration of edgecore can be changed after the join node to support quic. We can add some flags in the join command to support edgehub to choose websocket or quic.

If we add a bool parameter --quic, when choosing the quic protocol for the keadm join, add  --quic=true, --cloudcore-ipport and only fill in the cloud node IP. However, if using the quic protocol, the default port is 10001, but using websocket, the default port is 10000. If users need to modify the port, they can modify edgecore. yml and cloudcore after the keadm join. Is this okay?

Reply:
> If we add a bool parameter --quic, when choosing the quic protocol for the keadm join, add --quic=true, --cloudcore-ipport and only fill in the cloud node IP. However, if using the quic protocol, the default port is 10001, but using websocket, the default port is 10000. If users need to modify the port, they can modify edgecore. yml and cloudcore after the keadm join. Is this okay?

We can add a flag like `hubType= websocket / quic` to the cloud `init` and edge `join` commands, this way we don't have to restart the server to support quic after changing the confguration.

Can we create a new issue to track it? @fisherxu

Reply:
> > If we add a bool parameter --quic, when choosing the quic protocol for the keadm join, add --quic=true, --cloudcore-ipport and only fill in the cloud node IP. However, if using the quic protocol, the default port is 10001, but using websocket, the default port is 10000. If users need to modify the port, they can modify edgecore. yml and cloudcore after the keadm join. Is this okay?
> 
> We can add a flag like `hubType= websocket / quic` to the cloud `init` and edge `join` commands, this way we don't have to restart the server to support quic after changing the confguration.
> 
> Can we create a new issue to track it? @fisherxu

Cloudcore can enable both websocket and quic at the same time. I suggest that both websocket and quic of Cloudcore can be enabled at the same time, which can accept the protocol of quic adopted by edge nodes and also accept the protocol of websocket. When edge node keadm join, if the user is configured with - quic=true, edecore adopts the quic protocol. If not configured, websocket is used, which can reduce the configuration parameters filled in by the user when keadm join.

Reply:
> Cloudcore can enable both websocket and quic at the same time. I suggest that both websocket and quic of Cloudcore can be enabled at the same time, which can accept the protocol of quic adopted by edge nodes and also accept the protocol of websocket. When edge node keadm join, if the user is configured with - quic=true, edecore adopts the quic protocol. If not configured, websocket is used, which can reduce the configuration parameters filled in by the user when keadm join.

It is a very good practice for the Cloudcore to support multiple network communication methods at the same time.

The multi-choice method of edge communication has better extensibility, and if there are more communication modes in the future, this flag `quic=true` will need to be adjusted again.

Reply:
> > Cloudcore can enable both websocket and quic at the same time. I suggest that both websocket and quic of Cloudcore can be enabled at the same time, which can accept the protocol of quic adopted by edge nodes and also accept the protocol of websocket. When edge node keadm join, if the user is configured with - quic=true, edecore adopts the quic protocol. If not configured, websocket is used, which can reduce the configuration parameters filled in by the user when keadm join.
> 
> It is a very good practice for the Cloudcore to support multiple network communication methods at the same time.
> 
> The multi-choice method of edge communication has better extensibility, and if there are more communication modes in the future, this flag `quic=true` will need to be adjusted again.

Okay, I have revised the PR and now added the hubType field in the keadm join.

Reply:
> > > Cloudcore can enable both websocket and quic at the same time. I suggest that both websocket and quic of Cloudcore can be enabled at the same time, which can accept the protocol of quic adopted by edge nodes and also accept the protocol of websocket. When edge node keadm join, if the user is configured with - quic=true, edecore adopts the quic protocol. If not configured, websocket is used, which can reduce the configuration parameters filled in by the user when keadm join.
> > 
> > 
> > It is a very good practice for the Cloudcore to support multiple network communication methods at the same time.
> > The multi-choice method of edge communication has better extensibility, and if there are more communication modes in the future, this flag `quic=true` will need to be adjusted again.
> 
> Okay, I have revised the PR and now added the hubType field in the keadm join.



> > Cloudcore can enable both websocket and quic at the same time. I suggest that both websocket and quic of Cloudcore can be enabled at the same time, which can accept the protocol of quic adopted by edge nodes and also accept the protocol of websocket. When edge node keadm join, if the user is configured with - quic=true, edecore adopts the quic protocol. If not configured, websocket is used, which can reduce the configuration parameters filled in by the user when keadm join.
> 
> It is a very good practice for the Cloudcore to support multiple network communication methods at the same time.
> 
> The multi-choice method of edge communication has better extensibility, and if there are more communication modes in the future, this flag `quic=true` will need to be adjusted again.

Okay, I have modified the PR again, and now I have added the hubType field in the keadm join, and changed the - cloudcore ipport parameter to - cloudcore ip. All you need to do is specify the IP. When selecting websocket, the port is 10000, and when selecting quic, the port is 100001.

Reply:
> > Cloudcore can enable both websocket and quic at the same time. I suggest that both websocket and quic of Cloudcore can be enabled at the same time, which can accept the protocol of quic adopted by edge nodes and also accept the protocol of websocket. When edge node keadm join, if the user is configured with - quic=true, edecore adopts the quic protocol. If not configured, websocket is used, which can reduce the configuration parameters filled in by the user when keadm join.
> 
> It is a very good practice for the Cloudcore to support multiple network communication methods at the same time.
> 
> The multi-choice method of edge communication has better extensibility, and if there are more communication modes in the future, this flag `quic=true` will need to be adjusted again.

Okay, I have modified the PR again, and now I have added the hubType field in the keadm join, and changed the - cloudcore ipport parameter to - cloudcore ip. You only need to specify the ip and no longer need to specify the port number. When selecting websocket, the port is 10000, and when selecting quic, the port is 10001.